python/pyspark/sql/tests.py

```diff
@@ -1508,6 +1508,23 @@ def test_toDF_with_schema_string(self):
         self.assertEqual(df.schema.simpleString(), "struct<value:int>")
         self.assertEqual(df.collect(), [Row(key=i) for i in range(100)])
 
+    # Regression test for invalid join methods when on is None, Spark-14761
+    def test_invalid_join_method(self):
+        df1 = self.sqlCtx.createDataFrame([("Alice", 5), ("Bob", 8)], ["name", "age"])
+        df2 = self.sqlCtx.createDataFrame([("Alice", 80), ("Bob", 90)], ["name", "height"])
+        self.assertRaises(AnalysisException, lambda: df1.join(df2, how="invalid-join-type"))
+
+        result = df1.join(df2, how="inner").select(df1.name, df2.height).collect()
```

Comment:
can we remove everything from this test onward? they are no longer testing invalid join methods.